tests/scripts/arp_responder.py

```diff
@@ -39,18 +44,20 @@ def __del__(self):
             self.socket.close()
 
     def bind(self):
-        self.socket = socket.socket(socket.AF_PACKET, socket.SOCK_RAW, socket.htons(self.ETH_P_ALL))
-        self.socket.bind((self.iface, 0))
-        self.socket.settimeout(self.RCV_TIMEOUT)
+        self.socket = scapy2.conf.L2listen(iface=self.iface)
 
     def handler(self):
-        return self.socket.fileno()
+        return self.socket
 
     def recv(self):
-        return self.socket.recv(self.RCV_SIZE)
+        sniffed = self.socket.recv()
+        pkt = sniffed[0]
+        str_pkt = str(pkt).encode("HEX")
+        binpkt = binascii.unhexlify(str_pkt)
+        return binpkt
```

Comment:
There's probably some sort of scapy API we could use here, like we've done for `bind` and `send`. That would be a lot more clear.

Reply:
backlog